### PR TITLE
Codecheck bug fix

### PR DESCRIPTION
**Pull request by TangLeDaily**
【Fix issue】#1365

【Modified content】
The codecheck problems in the existing code have been modified, which are mainly summarized into the following categories;
| mudule | codecheck |
| --- | --- |
| sermant-agentcore-core | Code that needs to use AccessController.doPrivileged for security permission checking |
| sermant-agentcore-core | RuntimeException should not be thrown directly |
| sermant-agentcore-core | Possible multi-thread conflict, double check lock required |
| sermant-agentcore-core | Unused import |
| sermant-agentcore-core | Javadoc issues |
| sermant-agentcore-core | Devil's numbers |
| sermant-agentcore-core | Space indentation |
| sermant-agentcore-core | Multiple Boolean expressions in conditional statements |
| sermant-agentcore-core | An extra comma is added to the last element when defining the array |
| sermant-agentcore-core | Lambda expressions should use 2 or more letters |
| sermant-agentcore-core | deprecated annotations should add doc comments |
| sermant-agentcore-core | Exception code block cannot be empty |
| sermant-agentcore-core | Rewriting compareTo should also rewrite hashCode and equals |
| sermant-agentcore-core | When the parent class overrides the method of Object, the subclass also needs to rewrite it |
| sermant-agentcore-core | return null problem |
| sermant-agentcore-core | Use EntrySet instead of KeySet when traversing Map |
| sermant-agentcore-core | After creating a thread through new Thread, you should use the setUncaughtExceptionHandler method to register a runtime exception Handler |
| sermant-agentcore-god | deprecated annotations should add doc comments |
| sermant-agentcore-god | Code that needs to use AccessController.doPrivileged for security permission checking |
| sermant-agentcore-god | RuntimeException should not be thrown directly |
| sermant-agentcore-implement | Javadoc issues |
| sermant-agentcore-implement | Space indentation |
| sermant-agentcore-implement | Unused import |
| sermant-agentcore-implement | When it is determined that the incoming variable is null, the value null should be passed directly instead of the empty variable |
| sermant-agentcore-implement | Static string application static final definition |
| sermant-agentcore-implement | The parent class constructor should not call methods that may be overridden by subclasses. Instead, use final or private to modify the method |
| sermant-agentcore-implement | Double check lock acquisition singleton should add volatile to the singleton |
| sermant-agentcore-implement | Input parameter variables should not be modified directly |
| sermant-agentcore-implement | Unused variables |
| sermant-agentcore-implement | Simple logic without using switch code blocks |
| sermant-agentcore-implement | Use EntrySet instead of KeySet when traversing Map |
| sermant-agentcore-premain | Log objects should be decorated with final |
| sermant-agentcore-premain | RuntimeException should not be thrown directly |
| sermant-backend | Space indentation |
| sermant-backend | The get and set method names of entity classes should be consistent with the variable names |
| sermant-backend | Javadoc issues |
| sermant-backend | Simple logic without using switch code block |
| sermant-backend | lambda expressions should use 2 or more characters |
| sermant-backend | Mold variables of enumeration classes should be capitalized |
| sermant-backend | The parent class constructor should not call methods that may be overridden by subclasses. Instead, use final or private to modify the method |
| sermant-backend | Delete empty public constructor |
| sermant-backend | Variable names should use 2 or more characters |
| sermant-backend | The close method should not be used within the try structure, but the try catch resources structure should be used |
| sermant-backend | Input parameter variables should not be modified directly |

[Use case description] No

[Self-test situation] 1. Local static check passed

[Scope of influence] None

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java [hunk 1]
```diff
@@ -69,21 +92,23 @@ public static void setField(Object obj, Field field, Object value) {
...
-            setterName = "set" + fieldName.substring(2);
-        } else {
-            final char head = fieldName.charAt(0);
-            setterName = "set" + (
-                    (head >= 'a' && head <= 'z') ? ((char) (head + 'A' - 'a')) + fieldName.substring(1) : fieldName);
+        String setterName;
+        final char head = fieldName.charAt(0);
+        setterName = "set" + (
+                (head >= 'a' && head <= 'z') ? ((char) (head + 'A' - 'a')) + fieldName.substring(1) : fieldName);
+        if ((type == boolean.class || type == Boolean.class) && fieldName.length() >= FIELD_NAME_MIN_LENGTH) {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 这块为啥不要if else了？这样写的话第一次对settername赋值一定会执行

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java [hunk 2]
```diff
@@ -129,15 +161,17 @@ public static Object getField(Object obj, Field field) {
...
-                fieldName.length() >= 3 && (fieldName.startsWith("is") || fieldName.startsWith("Is")) &&
-                fieldName.charAt(2) >= 'A' && fieldName.charAt(2) <= 'Z') {
-            getterName = "is" + fieldName.substring(2);
-        } else {
-            final char head = fieldName.charAt(0);
-            getterName = "get" + (
-                    (head >= 'a' && head <= 'z') ? ((char) (head + 'A' - 'a')) + fieldName.substring(1) : fieldName);
+        String getterName;
+        final char head = fieldName.charAt(0);
+        getterName = "get" + (
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 同上

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigValueUtil.java [hunk 3]
```diff
@@ -121,7 +121,8 @@ public class ConfigValueUtil {
             },
             (key, argsMap, provider) -> System.getenv(key),
             (key, argsMap, provider) -> System.getProperty(key),
-            (key, argsMap, provider) -> provider == null ? null : provider.getFixedValue(key)
+            (key, argsMap, provider) -> provider == null ? null : (
+                    StringUtils.isEmpty(provider.getFixedValue(key)) ? null : provider.getFixedValue(key))
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigValueUtil.java]: 字符串为空也要传null吗

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/classloader/PluginClassFinder.java [hunk 4]
```diff
@@ -64,8 +72,8 @@ public Class<?> loadSermantClass(String name) throws ClassNotFoundException {
                 if (clazz != null) {
                     return clazz;
                 }
-            } catch (ClassNotFoundException ignore) {
-                // ignore
+            } catch (ClassNotFoundException e) {
+                LOGGER.log(Level.WARNING, "load sermant class failed, msg is {0}", e.getMessage());
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/classloader/PluginClassFinder.java:76]: catch块里error级别的日志吧

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/classloader/PluginClassLoader.java [hunk 5]
```diff
@@ -94,9 +102,10 @@ public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundExce
             if (clazz == null) {
                 try {
                     clazz = super.loadClass(name, resolve);
-                } catch (ClassNotFoundException ignored) {
+                } catch (ClassNotFoundException e) {
                     // 捕获类找不到的异常，下一步会进入localLoader中去加载类
                     // ignored
+                    LOGGER.log(Level.WARNING, "load class failed, msg is {0}", e.getMessage());
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/classloader/PluginClassLoader.java:108]: 同上，下面的也排查下

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/LabelGroupUtils.java [hunk 6]
```diff
@@ -148,8 +149,8 @@ public static Map<String, String> resolveGroupLabels(String group) {
                     LOGGER.warning(String.format(Locale.ENGLISH, "Invalid label [%s]", label));
                 }
             }
-        } catch (UnsupportedEncodingException ignored) {
-            // ignored
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "UnsupportedEncodingException, msg is {0}.", e.getMessage());
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/LabelGroupUtils.java:153]: 日志级别

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/ThreadFactoryUtils.java [hunk 7]
```diff
@@ -70,6 +76,13 @@ private String prefix(String threadName, int factoryId) {
     public Thread newThread(Runnable job) {
         String newThreadName = createThreadName();
         Thread thread = new Thread(job, newThreadName);
+        thread.setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
+            @Override
+            public void uncaughtException(Thread t, Throwable e) {
+                LOGGER.log(Level.WARNING, "uncaughtException, thread is {0}, msg is {1}", new String[]{t.getName(),
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/ThreadFactoryUtils.java:82]: 这块考虑下是error还是warning

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/common/ConfigFieldKey.java [hunk 8]
```diff
@@ -16,18 +16,16 @@
...
-
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
  * 通用字段键注解
- * <p>用于修饰配置对象的属性，与{@link ConfigTypeKey}一并构建配置信息键
+ * <p>用于修饰配置对象的属性，与{ConfigTypeKey一并构建配置信息键
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/common/ConfigFieldKey.java]: 缺少右括号

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/dynamicconfig/kie/client/kie/KieListenerWrapper.java [hunk 9]
```diff
@@ -153,22 +152,16 @@ private void processAllListeners(DynamicConfigEvent event, VersionListenerWrappe
...
-                    processInit(latestData, versionListener, event);
-                } else {
-                    versionListener.listener.process(event);
-                }
-                versionListener.version = currentVersion;
-            } catch (Exception ex) {
-                LOGGER.log(Level.WARNING, String.format(Locale.ENGLISH,
-                        "Process config data failed, key: [%s], group: [%s]",
-                        event.getKey(), this.group), ex);
+            if (versionListener.version > currentVersion) {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/dynamicconfig/kie/client/kie/KieListenerWrapper.java]: 这块不会有异常抛出了吗

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java [hunk 10]
```diff
@@ -69,21 +92,23 @@ public static void setField(Object obj, Field field, Object value) {
...
+        setterName = "set" + (
+                (head >= 'a' && head <= 'z') ? ((char) (head + 'A' - 'a')) + fieldName.substring(1) : fieldName);
+        if ((type == boolean.class || type == Boolean.class) && fieldName.length() >= FIELD_NAME_MIN_LENGTH) {
+            if ((fieldName.startsWith(BOOLEAN_FUNCTION_PREFIX_LOWERCASE)
+                    || fieldName.startsWith(BOOLEAN_FUNCTION_PREFIX_UPPERCASE))
+                    && fieldName.charAt(FIELD_NAME_CHECK_INDEX) >= 'A'
+                    && fieldName.charAt(FIELD_NAME_CHECK_INDEX) <= 'Z') {
+                setterName = "set" + fieldName.substring(FIELD_NAME_CHECK_INDEX);
+            }
         }
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 判断条件合理组合抽出判断函数来降低圈复杂度

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/common/ConfigFieldKey.java (hunk 8 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/common/ConfigFieldKey.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java (hunk 1 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java (hunk 2 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigValueUtil.java (hunk 3 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigValueUtil.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/ThreadFactoryUtils.java (hunk 7 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/ThreadFactoryUtils.java:82]: 统一为warning了

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java (hunk 10 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 已修改

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/dynamicconfig/kie/client/kie/KieListenerWrapper.java (hunk 9 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/com/huaweicloud/sermant/implement/service/dynamicconfig/kie/client/kie/KieListenerWrapper.java]: 已修改

**Review by daizhenyu — DISMISSED** [review]

**Review by lilai23 — DISMISSED** [review]

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java [hunk 11]
```diff
@@ -130,10 +159,9 @@ public static Object getField(Object obj, Field field) {
 
     private static Method getGetter(Class<?> cls, String fieldName, Class<?> type) {
         final String getterName;
-        if ((type == boolean.class || type == Boolean.class) &&
-                fieldName.length() >= 3 && (fieldName.startsWith("is") || fieldName.startsWith("Is")) &&
-                fieldName.charAt(2) >= 'A' && fieldName.charAt(2) <= 'Z') {
-            getterName = "is" + fieldName.substring(2);
+        if ((type == boolean.class || type == Boolean.class) && fieldName.length() >= FIELD_NAME_MIN_LENGTH
+                && checkFieldName(fieldName)) {
+            getterName = "is" + fieldName.substring(FIELD_NAME_CHECK_INDEX);
```
Comment (luanwenfei-venus) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 这里的is上面有常量可以用

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/NetworkUtils.java [hunk 12]
```diff
@@ -59,7 +60,7 @@ public static List<String> getAllNetworkIp() {
         try {
             Enumeration<NetworkInterface> netInterfaces = NetworkInterface.getNetworkInterfaces();
             if (netInterfaces == null) {
-                throw new RuntimeException("netInterfaces is null");
+                throw new NullPointerException("netInterfaces is null");
```
Comment (luanwenfei-venus) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/NetworkUtils.java]: 这个不要抛出空指针异常，自定义一个异常吧

File: sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java [hunk 13]
```diff
@@ -50,25 +50,17 @@ private GzipUtils() {
...
         } catch (IOException e) {
-            LOGGER.error("Exception occurs when compress. Exception info: {}", e);
-        } finally {
-            try {
-                baos.close();
-                bais.close();
-            } catch (IOException e) {
-                LOGGER.error("Exception occurs when close IOStream. Exception info: {}", e);
-            }
+            LOGGER.error("Exception occurs when compress or close IOStream. Exception info: {}", e);
```
Comment (lilai23) [inline sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java]: IOException 不要直接打印，子类中的FileNotFoundException属于敏感异常

File: sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java [hunk 14]
```diff
@@ -126,7 +118,7 @@ public static byte[] decompress(byte[] data) {
                 LOGGER.error("Exception occurs when close IOStream. Exception info: {}", e);
```
Comment (lilai23) [inline sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java]: IOException 不打印堆栈

File: sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/AgentLauncher.java [hunk 15]
```diff
@@ -159,11 +160,11 @@ private static void executeCommand(String artifact, String command) {
     private static void appendGodLibToBootStrapClassLoaderSearch(Instrumentation instrumentation) {
         final File bootstrapDir = new File(PathDeclarer.getGodLibPath(PathDeclarer.getAgentPath()));
         if (!bootstrapDir.exists() || !bootstrapDir.isDirectory()) {
-            throw new RuntimeException("God directory is not exist or is not directory.");
+            throw new LauncherException("God directory is not exist or is not directory.");
```
Comment (luanwenfei-venus) [inline sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/AgentLauncher.java]: LauncherException有点太宽泛，这里主要是核心目录检查的异常

File: sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/utils/LoggerUtils.java [hunk 16]
```diff
@@ -29,21 +29,13 @@
  * @since 2023-07-20
  */
 public class LoggerUtils {
-    private static Logger logger;
+    private static final Logger logger = initLogger();
```
Comment (luanwenfei-venus) [inline sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/utils/LoggerUtils.java]: static final的变量名应该需要全大写

File: sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java (hunk 14 above)
Comment (TangLeDaily) [reply sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java]: 已修改

File: sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java (hunk 13 above)
Comment (TangLeDaily) [reply sermant-backend/src/main/java/com/huaweicloud/sermant/backend/util/GzipUtils.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java (hunk 11 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/utils/ConfigFieldUtil.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/NetworkUtils.java (hunk 12 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/utils/NetworkUtils.java]: 已修改

File: sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/AgentLauncher.java (hunk 15 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/AgentLauncher.java]: 已修改

File: sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/utils/LoggerUtils.java (hunk 16 above)
Comment (TangLeDaily) [reply sermant-agentcore/sermant-agentcore-premain/src/main/java/com/huaweicloud/sermant/premain/utils/LoggerUtils.java]: 已修改

**Review by luanwenfei-venus — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**